### DISCUSSION TITLE
TiXI Matlab support

### DISCUSSION THREAD

**Discussion by MarAlder** [DLR-SC/tixi#222] [General]
I would like to open an issue on behalf of some TU-Delft colleagues who are trying to use the Matlab bindings. As it is a bit difficult for me to reproduce Matlab problems as I am not a Matlab user, maybe someone in this forum can help?

> I’m currently trying to read cpacs file using matlab, with tigl/tixi libraries. I followed the tutorial online, where it says a precompiled mex file is provided. However, when I was trying to open xml file, matlab returns me an error saying:
> Error using tixiOpenDocument (line 11)
> Invalid MEX-file 'C:\Program Files\TIXI 3.3.0\share\tixi3\matlab\tixi3_matlab.mexw64': The specified procedure could not be found.
> 
> The tigl matlab example also returns an error when I was trying to open the provided xml.
> Error using matlab_demo (line 22)
> Error opening cpacs file with TiXI
> 
> I have made sure both tigl and tixi matlab libraries have been added to the search path. I have tried using both matlab 2021b and 2023b. I’m not very good with programming/coding, maybe I made some silly mistakes. I spent some time to troubleshoot but cannot resolve the issue. Could you guys please provide me some insights on this? 
> 
> Thank you so much for your time.
> 
> Best Regards,
>

**Comment by joergbrech** [comment]
There is a related discussion at the TiGL repo: https://github.com/DLR-SC/tigl/discussions/914

It is hard to debug this problem remotely. But in general, to work with tixi and tigl from Matlab in Windows it should suffice to

  -  have `tixi3.dll` and `tigl3.dll` in your Windows path
  -  have `tixi3_matlab.mexw64` and `tigl_matlab.mexw64` in your Matlab search path.

> I have made sure both tigl and tixi matlab libraries have been added to the search path. 

Have they been added to the windows `PATH` environment variable or Matlab search path? As far as I know, Matlab searches for Matlab files and precompiled mex files in the Matlab search path, but **not** shared libraries.

**Reply by SW-TUD** [reply]
Thank you so much for the reply. I have verified that both dll and mex files for tixi3 and tigl3 are in the windows PATH and Matlab search path. The problem still exists. The error messages I received are also the same with the ones posted in the other discussion:

Error using tixiOpenDocument (line 11)
Invalid MEX-file 'C:\Program Files\TIXI 3.3.0\share\tixi3\matlab\tixi3_matlab.mexw64': The specified
procedure could not be found.

Error in untitled (line 11)
Handle = tixiOpenDocument('C:\Program Files\TIGL 3.3.0\share\doc\tigl3\examples\simpletest.cpacs.xml') 

The issues on the Matlab side seems not being resolved from that discussion either. I will dig a little bit more, and update here if I find any solutions.

**Reply by HHOPs** [reply]
I remember, having similar problems in the past. Therefore I would propose two options: 1. Quick check, if the relevant dependencies for your mexw64 binary can be found - http://www.dependencywalker.com/, 2. Compile the binaries yourself (MS Visual Studio, MinGW) should do.
